### DISCUSSION TITLE
Can not reboot when watchdog triggered in AMP architecture

### DISCUSSION THREAD

**Discussion by xing-git** [polarfire-soc/polarfire-soc-documentation#196] [HSS and U-Boot] [answered]
My system use AMP arch,HSS runs in E51 in l2lim,other APP run in U54 in DDR.The 5 projects compile separately.
When power on, HSS boot U54s and all run well.
When a U54 watchdog triggered,E51 try to reboot U54 ,when U54 boot machine states run in "BOOT_SETUP_PMP_COMPLETE"，its cann't continue run because no IPI ack.
![捕获](https://user-images.githubusercontent.com/20128646/187579104-31ba8421-3fb5-432a-b768-556a781cd372.JPG)

I think the reason is: when firstly boot,U54 runs crt.s in HSS,software int handle is "HSS_U54_HandleIPI",it can response IPI msg. After u54 boot,  it runs the soc_u54 software in DDR,the software int handle is "Software_hX_IPQHandler" and there is no IPI module in u54 project.So cant response IPI when HSS try to send IPI msg to a U54 hart.
I want to know if the reason I think is right, and how can I reboot a u54 hart when systerm is running.

**Comment by griffini** [comment, marked as answer]
Hi @xing-git , thanks for raising this.

Context-aware reboot is coming with the very next release of the HSS, so that should solve your issue.

When you say the HSS is still running out of L2LIM are you using a particular old version of the HSS?

**Reply by xing-git** [reply]
Thank you for your answer and I am waiting for the newer release.
Yes I am use the older version of HSS.I had tried to use the version release on 2022.07.02.But I meet some trouble.One trouble is about IPI use IHC,this is mentioned in another disscussion. Another trouble is often run into trap.
But the detail I can't remenber clearly,Because that time I am busy for other things ,so directly roll back to old version.
I will try to use new release recently
